### PR TITLE
Update to Smack 4.2

### PR DESCRIPTION
**Pull request by ibauersachs**

### CODE REVIEW

File: src/main/java/org/jitsi/jigasi/xmpp/CallControl.java [hunk 1]
```diff
@@ -146,99 +159,91 @@ public CallControl(ConfigurationService config)
...
+     * @return an {@link IQ} stanza which represents the response to the
+     * specified request.
+     */
+    public IQ handleHangUp(HangUp iq)
+        throws CallControlAuthorizationException
+    {
+        checkAuthorized(iq);
 
-                SipGatewaySession session = sipGateway.getSession(callResource);
+        // FIXME: 23/07/17 Transcription sessions should also be able
```
Comment (bbaldino) [inline src/main/java/org/jitsi/jigasi/xmpp/CallControl.java]: anything here that needs to be addressed before merging?

Comment (ibauersachs) [reply src/main/java/org/jitsi/jigasi/xmpp/CallControl.java]: This must be a mismatched merge, the note isn't from me.

Comment (ibauersachs) [reply src/main/java/org/jitsi/jigasi/xmpp/CallControl.java]: Actually not a mismatched merge. The note was there before, just at a different position (in the if/multi-else type check of handleIQ).

**Comment by nikvaessen** [comment]
This PR should be ready to be reviewed again.

**Comment by damencho** [comment]
Me and @nikvaessen we've been testing for a while with this branch, so if no one objects I will merge it tomorrow morning (CDT).

**Pull request merged**